BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +56,560 @@ public struct AttributedLabel: Element, Hashable {
         )
     }
 
-    private func update(label: LabelView) {
-        label.attributedText = attributedText
-        label.numberOfLines = numberOfLines
-        label.textRectOffset = textRectOffset
-        label.isAccessibilityElement = isAccessibilityElement
-        updateAccessibilityTraits(label)
-    }
-
     public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
         LabelView.describe { config in
             config.frameRoundingBehavior = .prioritizeSize
-            config.apply(update)
+            config.apply { view in
+                view.update(model: self, urlHandler: context.environment.urlHandler, isMeasuring: false)
+            }
         }
     }
+}
+
+// MARK: - Models
+
+extension AttributedLabel {
+    /// Types of data that can be detected and automatically turned into links.
+    ///
+    /// Corresponds to `NSTextCheckingResult.CheckingType` types.
+    ///
+    public enum LinkDetectionType: Equatable, Hashable {
+        /// Detect date strings. Tapping a date opens the calendar to that date.
+        case date
+        /// Detect addresses. Tapping the address opens Maps with that address.
+        case address
+        /// Detect URLs. Tapping the link opens the URL.
+        case link
+        /// Detect phone numbers. Tapping the phone number prompts the user to call it.
+        case phoneNumber
 
-    private func updateAccessibilityTraits(_ label: UILabel) {
-        if let traits = accessibilityTraits {
-            var union = label.accessibilityTraits.union(UIAccessibilityTraits(with: traits))
-            // UILabel has the `.staticText` trait by default. If we explicitly set `.updatesFrequently` this should be removed.
-            if traits.contains(.updatesFrequently) && label.accessibilityTraits.contains(.staticText) {
-                union.subtract(.staticText)
+        var checkingType: NSTextCheckingResult.CheckingType {
+            switch self {
+            case .date: return .date
+            case .address: return .address
+            case .link: return .link
+            case .phoneNumber: return .phoneNumber
             }
-            label.accessibilityTraits = union
         }
     }
 }
 
 extension AttributedLabel {
+    /// Allows adding URL links or tap handlers to `AttributedLabel` ranges.
+    ///
+    /// The range must be a range within the attributed string provided to the label.
+    ///
+    public struct Link: Equatable, Hashable {
+        private let identifier: Int
+
+        var type: LinkType
+
+        /// The range of text over which to render the link.
+        public var range: NSRange
+
+        /// Create a link with a URL.
+        ///
+        /// - Parameters:
+        ///   - url: The url to open.
+        ///   - range: The range of text over which to render the link.
+        public init(url: URL, range: NSRange) {
+            identifier = url.hashValue
```

Comment:
(fine to keep as is): Any reason not to just use `UUID()` here as well? It seems a little weird that two links in different locations (with matching URLs) would share an `identifier`.

Reply:
This is all deleted